### ISSUE TITLE
C# Mailbox setting automatic replies update failing with OData error

### ISSUE THREAD

**Issue by Talhakh** [microsoftgraph/microsoft-graph-docs#5007]
Issue: 
Trying the C# request for updating the mailbox settings for automatic replies using the Graph client but the code keeps on throwing the exception
Code: ErrorInvalidRequest
Message: The OData request is not supported.
Inner error

Article:
https://docs.microsoft.com/en-us/graph/api/user-update-mailboxsettings?view=graph-rest-1.0&tabs=csharp#example

**Comment by MIchaelMainer** [comment]
Is this still an issue for anyone? Can you provide a repro?

**Comment by djmrky** [comment]
Yes, I have an issue just like this today. Trying to update the mailbox settings for a user and getting an error.
The difference from the docs is that I'm trying to do this not for "me", but for other user. The permissions are set correctly. 

`  
Microsoft.Graph.User user = new Microsoft.Graph.User();
  //setup mailbox settings object
                    Microsoft.Graph.MailboxSettings mailboxSettings = new Microsoft.Graph.MailboxSettings
                    {
                        AutomaticRepliesSetting = new Microsoft.Graph.AutomaticRepliesSetting()
                        {
                            Status = AutomaticRepliesStatus.AlwaysEnabled,
                            InternalReplyMessage = "something",
                            ExternalReplyMessage = "something",
                        }
                    };
user.MailboxSettings = mailboxSettings;
  await _graphServiceClient
                                    .Users[<userId>]
                                    .Request()
                                    .UpdateAsync(user);
`

getting an error:
Code: ErrorInvalidRequest
Message: The OData request is not supported.
Inner error:
	AdditionalData:
	request-id: fcc4c279-2280-4c96-8ec2-a8c8be1a3623
	date: 2020-04-17T09:04:21
ClientRequestId: fcc4c279-2280-4c96-8ec2-a8c8be1a3623

**Comment by jasonjoh** [comment]
Are you using an application token or a user token?

**Comment by djmrky** [comment]
I'm getting the token via http request like this:

  string getTokenUrl = $"https://login.microsoftonline.com/" + this._tenant.TenantId + "/oauth2/v2.0/token";

                Dictionary<string, string> bodyParameters = new Dictionary<string, string>();

                bodyParameters.Add("client_id", this._clientId);
                bodyParameters.Add("client_secret", this._appKey);
                bodyParameters.Add("grant_type", "client_credentials");
                bodyParameters.Add("scope", "https://graph.microsoft.com/.default");


The get of the Mailbox settings is wokring correctly, but the set isn't

This is the permissions that's been granted in the enterprise application:
Graph | Read and write all user mailbox settings

**Comment by jasonjoh** [comment]
Ok, so an application token (`client_credentials`). I just tried it with Postman and it worked fine for me, so it may be something about how the SDK builds the request. Will test and post findings.

**Comment by jasonjoh** [comment]
Yes, I reproduce this with the SDK. Fiddler reveals the problem. The SDK is sending a PATCH to `/users/{user-id}`, but per the docs it should be a PATCH to `/users/{user-id}/mailboxsettings`. This looks to be a bug with the SDK, so I've opened an issue there (https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/673).

To work around this, you can use the SDK's underlying HTTP provider. I've created a Gist with sample code: https://gist.github.com/jasonjoh/f4d9ec4c9eae01c23bad12eadf409088

**Comment by djmrky** [comment]
Great catch ... For the time being, I've switched (back) to the REST endpoints for maintaining the mailbox since it was a time-critical issue, but will implement it with the SDK once the patch comes out.

**Comment by plp11** [comment]
Same problem

**Comment by ssougnez** [comment]
I'm currently using EWS to update the mailbox settings, however, Microsoft announced that EWS will be decommissioned second half of this year, so I'm trying to migrate to msgraph.

It's crazy that this issue is opened for almost two years and is still not solved while @jasonjoh gave the solution for it.
